### PR TITLE
Add cd-rom 'empty tray' regression test

### PR DESCRIPTION
**Pull request by mcgov**
Adding a regression test to check virtual cd/dvd tray reports 'empty' after initial provisioning is complete.
A recent bug would cause the tray to report CDS_DISK_OK which means 'the drive has a disk inside and it's ready to use' even though there is no ISO/CD/DVD in the 'tray'. 
This test uses a small portable C program to check that the behavior is fixed after the initial provisioning process; all linux distros will support these commands; they've been around as long as CD-ROM support has.

### CODE REVIEW

File: microsoft/testsuites/cdrom/cdrom.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    )
+
+    @TestCaseMetadata(
+        description="""
+            Test to check the installation ISO is unloaded
+            after provisioning and rebooting a new VM.
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            supported_os=[Linux], supported_platform_type=[AZURE]
```
Comment (squirrelsc) [inline microsoft/testsuites/cdrom/cdrom.py:37]: It doesn't need to specify Linux, it's default settings.

File: microsoft/testsuites/cdrom/cdrom.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        # and run the test if the device is present.
+        result = node.execute(
+            cmd=str(node_exec_path),
+            shell=True,
+            sudo=True,
+        )
+        output = result.stdout.strip()
+
+        # first check for failure to open the cdrom device
+        assert_that(result.exit_code).described_as(
```
Comment (squirrelsc) [inline microsoft/testsuites/cdrom/cdrom.py:76]: I feel it doesn't need to assert for -1 separately. The raw output should be able to explain the error.

File: microsoft/testsuites/cdrom/cdrom.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    # CD drive status constants from linux uapi
+    # see https://github.com/torvalds/linux/blob/master/include/uapi/linux/cdrom.h#L417
+    _expected_device_status = (
+        "CDS_NO_DISC"  # drive tray is closed, no disk in the drive
+    )
+
+    @TestCaseMetadata(
+        description="""
+            Test to check the installation ISO is unloaded
+            after provisioning and rebooting a new VM.
```
Comment (LiliDeng) [inline microsoft/testsuites/cdrom/cdrom.py:33]: what is the impact if the iso is not unloaded?

Comment (mcgov) [reply microsoft/testsuites/cdrom/cdrom.py:33]: I am not aware of the full scope of the issue, I believe Michael pointed the problem out and the host OS team has a fix ready. I cleaned up the very basic test that was shared with me and implemented it as a LISA test for regression checks going forward.
I would imagine any program which waits for the tray to be empty would get stuck forever though. Think 'Remove install ISO...'

File: microsoft/testsuites/cdrom/cdrom.py (hunk 1 above)
Comment (mcgov) [reply microsoft/testsuites/cdrom/cdrom.py:37]: Does supported_os=[Linux] not enforce that other OS's are unsupported?
I've only tested on Linux and don't see the need to support BSD. It was a host side fix so whichever OS checks the status of the drive should get the same result.

File: microsoft/testsuites/cdrom/cdrom.py (hunk 2 above)
Comment (mcgov) [reply microsoft/testsuites/cdrom/cdrom.py:76]: I disagree, it's a subtle distinction but not being able to open /dev/cdrom implies a test bug (priviledges or distro differences) while the second assertion definitively implies a host bug. 

Having different error messages for each is a time saver when you're triaging through hundreds of tests. This is a bigger issue but there are a lot of error messages that LISA raises that I wish were more specific about _what should be done_ when you see the error raised.

**Comment by LiliDeng** [comment]
please fix compile issue on mariner @mcgov

**Comment by mcgov** [comment]
> please fix compile issue on mariner @mcgov

Fixed! Thanks for pointing this out. I've decided to only support Mariner and Debian to reduce future build-breaking headaches. I've also added a check for a failed build and throw a specific error message for that case so it should be easier to triage in the future.

**Review by LiliDeng — APPROVED** [review]
